docs/user/app_feature_intended.md

```diff
@@ -104,3 +105,35 @@ This is exactly the same concept as described in [Backup Repository Matching Rul
 ## Data
 
 The data provided while rendering the configuration of a device is described in the [SoT Aggregation](./app_feature_sotagg.md) overview.
+
+## Render Intended Configuration as a final push artifact
+
+The Intended Configuration Job doesn't produce a final configuration artifact ready to use to update a network device. You should understand it as the "intended" running configuration, because the main purpose is to be compared against the current running configuration (retrieved in the "backup" job).
+
+Aside of enabling the "compliance" feature, there are some other limitations on the intended configuration:
+
+- Because the intended configuration is stored in the Database, and in an external Git repository, it SHOULD NOT contain any secret (or derivative).
+- The format of the running configuration is not exactly the same as the configuration to push in some devices.
+
+However, Golden Config following the purpose to become the cornerstone configuration management application is providing an advanced feature to render the intended configuration in the final format your device is expecting to.
+
+In the `Device` detail view, and in the API endpoint `config-to-push`, you can obtain the final configuration artifacts for the devices. By default, there is a function to render secrets using a `get_secrets` filter, and also a custom hook can be attached in the post-processing process.
+
+### Get Secrets filter
+
+The `get_secrets` filter uses Nautobot Secrets object to render as text. Because this render happens not in the first one to generate the intended configuration, but on a second one, you must use the `{% raw %}` Jinja syntax to avoid being processed on the first one. For example:
+
+```jinja
+password {{ id | get_secret("dcim.Device", "password") | encrypt_type5 }}
```

Comment:
I was expecting to see `{% raw %}` here

Reply:
the filter is not aware of how it would be used, in 2 rendering steps. I agree that it should be in the overall user documentation, but I don't think so for a specific filter